tensorflow/python/keras/backend.py

```diff
@@ -3897,7 +3897,7 @@ def binary_crossentropy(target, output, from_logits=False):
 
       # Compute cross entropy from probabilities.
       bce = target * math_ops.log(output + epsilon())
-      bce += (1 - target) * math_ops.log(1 - output + epsilon())
+      bce += (1 - target) * math_ops.log1p(epsilon() - output)
```

Comment:
@ebrevdo Thanks for pointing out the failing tests. This change makes the tests fail although it should be mathematically equivalent.